ui/src/overridden/frontsite/Document/DocumentDetails/DocumentItemBody.js

```diff
@@ -1,30 +1,50 @@
+import React from "react";
 import {
   DocumentItemBody,
   invenioConfig,
+  InfoPopup,
 } from "@inveniosoftware/react-invenio-app-ils";
 import _get from "lodash/get";
+import _isEmpty from "lodash/isEmpty";
 import { parametrize } from "react-overridable";
 import { renderCallNumber, shelfLinkComponent } from "../../../utils";
 
 function renderShelflink(item, documentDetails) {
   const itemStatus = _get(item, "circulation.state");
 
-  // If item is on loan, don't hyperlink the shelf
-  const cannotCirculate =
-    invenioConfig.CIRCULATION.loanActiveStates.includes(itemStatus);
-  const itemOnShelf = ["CAN_CIRCULATE", "FOR_REFERENCE_ONLY"].includes(
+  // If item is not on loan, hyperlink the shelf
+  const canCirculateItem =
+    !invenioConfig.CIRCULATION.loanActiveStates.includes(itemStatus);
+  const itemIsOnShelf = ["CAN_CIRCULATE", "FOR_REFERENCE_ONLY"].includes(
     _get(item, "status")
   );
 
   const shelfNumber = _get(item, "shelf");
   const title = _get(documentDetails, "metadata.title");
   const callNumber = renderCallNumber(item);
 
+  var callNumberComponent = <> {callNumber}</>;
+  if (_isEmpty(shelfNumber) || shelfNumber === "") {
+    // If shelfNumber is empty, show info popup regardless of availability
+    callNumberComponent = (
+      <InfoPopup message="Please request online or ask at the Library desk.">
+        {" "}
+        {callNumber}
+      </InfoPopup>
+    );
+  }
+
   const itemShelf =
-    cannotCirculate && itemOnShelf
-      ? shelfNumber
-      : shelfLinkComponent(shelfNumber, title, callNumber);
-  return itemShelf;
+    canCirculateItem && itemIsOnShelf && !_isEmpty(shelfNumber) && shelfNumber !== ""
```

Comment:
isEmpty should already take care of empty string
```suggestion
    canCirculateItem && itemIsOnShelf && !_isEmpty(shelfNumber) 
```